src/popup/router/pages/Send.vue

```diff
@@ -201,26 +202,48 @@ export default {
     async fetchFee() {
       await this.$watchUntilTruly(() => this.sdk);
       this.fee = await calculateFee(
-        this.token === 0 ? TX_TYPES.txSign : TX_TYPES.contractCall,
-        this.sdk.Ae.defaults,
+        !this.selectedToken ? TX_TYPES.txSign : TX_TYPES.contractCall,
+        !this.selectedToken
+          ? this.sdk.Ae.defaults
+          : {
+              ...this.sdk.Ae.defaults,
+              callerId: this.account.publicKey,
+              contractId: this.selectedToken.contract,
+            },
       );
     },
     setTxDetails(tx) {
+      if (tx.tx.type === 'ContractCallTx') {
+        this.successTx.amount = convertToken(
+          tx.amount,
+          -this.availableTokens[tx.contractId].decimals,
+        );
+        this.successTx.token = tx.contractId;
+        this.successTx.to = tx.recipientId;
+        this.successTx.from = tx.callerId;
+        this.successTx.hash = tx.hash;
+        return;
+      }
       this.successTx.amount = parseFloat(tx.tx.amount / 10 ** 18).toFixed(3);
       this.successTx.to = tx.tx.recipientId;
       this.successTx.from = tx.tx.senderId;
       this.successTx.hash = tx.hash;
     },
     async send() {
-      const amount = aeToAettos(this.form.amount);
+      const amount = !this.selectedToken
+        ? aeToAettos(this.form.amount)
+        : convertToken(this.form.amount, this.selectedToken.decimals);
       const receiver = this.form.address;
       const calculatedMaxValue = this.balance > this.fee.max ? this.balance - this.fee.max : 0;
       let errorModalType = '';
       if (receiver === '' || (!checkAddress(receiver) && !chekAensName(receiver))) {
         errorModalType = 'incorrect-address';
       }
       if (this.form.amount <= 0) errorModalType = 'incorrect-amount';
-      if (calculatedMaxValue - this.form.amount <= 0 && this.token === 0) {
+      if (
+        (this.selectedToken && calculatedMaxValue <= 0) ||
+        calculatedMaxValue - this.form.amount <= 0
```

Comment:
Does it make sense to compare ae and token amounts? On tokens sending we should ensure that:
- the user's ae balance is bigger or equal than transaction fees
- the user's token balance is bigger or equal than the number of tokens he wants to send

Reply:
This is covered here. 
- The first check is if we have selected a FT and checks if our Ae balance can cover the fees.
 - The second check ensures our balance in AE is bigger than the filled number in the input - so we need to have funds.

Reply:
>- The second check ensures our balance in AE is bigger than the filled number in the input - so we need to have funds.

Second check should actually check if the FT balance of the user as @davidyuk suggested - `contract.methods.balance('ak_...')` * should be `>` than the amount the user wants to send. 

P.S. keep in mind to take care of FT decimal piceces.

Reply:
I'm trying to send tokens with 0 ae balance, but I don't see the `insufficient-balance` modal.